platforms/core-execution/execution/src/main/java/org/gradle/internal/execution/steps/IncrementalResolveCachingStateStep.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.execution.steps;
+
+import org.gradle.caching.internal.controller.BuildCacheController;
+import org.gradle.internal.execution.UnitOfWork;
+import org.gradle.internal.execution.caching.CachingState;
+import org.gradle.internal.hash.HashCode;
+
+import java.util.Objects;
+import java.util.Optional;
+
+public class IncrementalResolveCachingStateStep<C extends IncrementalChangesContext> extends AbstractResolveCachingStateStep<C> {
```

Comment:
🤔 While there would be benefits to having `Incremental` / `NonIncremental` be a prefix of the step name always, we currently don't do that with the others, and instead aim for a more readable name. I think we should go with `ResolveIncrementalCachingStateStep` here as well.

Reply:
Renamed.